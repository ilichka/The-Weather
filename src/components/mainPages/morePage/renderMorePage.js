import "./index.scss";
import * as waterDropIcon from "../../../assets/img/waterdrop.svg";
import * as windIcon from "../../../assets/img/wind.svg";
import * as pressureIcon from "../../../assets/img/pressure.svg";
import * as locationIcon from "../../../assets/img/location.svg";

const renderMorePage = (obj) => {
  if (Object.keys(obj).length) {
    const { city, country, temp, humidity, wind, icon } =
      obj[Object.keys(obj)[0]];
    return `
  <div class="ur-location">
          <img class="location-icon" src="${locationIcon.default}" alt="location"/>
          <span class="txt">Your location now</span>
        </div>
        <div class="place-name">${city}, ${country}</div>
        <img class="status-icon" src="http://openweathermap.org/img/wn/${icon}.png" alt="icon"/>
        <div class="status">Moonlight</div>
        <div class="temp">${temp}Â°</div>
        <div class="info-row">
          <div class="humidity-info">
            <img class="humidity-icon" src="${waterDropIcon.default}" alt="water drop"/>
            <span class="humidity">${humidity}%</span>
          </div>
          <div class="wind-info">
            <img class="wind-icon" src="${windIcon.default}" alt="wind"/>
            <span class="wind-speed">${wind}km/h</span>
          </div>
          <div class="pressure-info">
            <img class="pressure-icon" src="${pressureIcon.default}" alt="pressure"/>
            <span class="pressure">0.43mBar</span>
          </div>
        </div>
        <div class="btn-rows">
          <div class="btn-row">
            <span class="name">Temperature</span>
            <span class="value">celcius</span>
            <div class="arrow">&gt;</div>
          </div>
          <div class="btn-row">
            <span class="name">Wind speed</span>
            <span class="value">km/s</span>
            <div class="arrow">&gt;</div>
          </div>
          <div class="btn-row">
            <span class="name">Source</span>
            <span class="value">openWeatherAPI</span>
            <div class="arrow">&gt;</div>
          </div>
        </div>
  `;
  }
  return `<span class="no-match">Your history is clean:c</span>`;
};

export default renderMorePage;
