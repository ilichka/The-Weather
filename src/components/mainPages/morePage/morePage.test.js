import renderMorePage from "./renderMorePage";

const data = {
  first: {
    city: "city",
    country: "country",
    temp: "temp",
    humidity: "humidity",
    wind: "wind",
    icon: "04d",
  },
};

describe("More page snapshot", () => {
  it("Renders correctly template with mock data object", () => {
    const template = renderMorePage(data);

    expect(template).toMatchSnapshot();
  });
});
