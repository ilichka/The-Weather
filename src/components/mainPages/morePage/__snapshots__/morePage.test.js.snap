// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`More page snapshot Renders correctly template with mock data object 1`] = `
"
  <div class=\\"ur-location\\">
  <svg class=\\"location-icon\\" enable-background=\\"new 0 0 64 64\\" version=\\"1.1\\" viewBox=\\"0 0 64 64\\" xml:space=\\"preserve\\" xmlns=\\"http://www.w3.org/2000/svg\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\"><g id=\\"Glyph\\"><path d=\\"M52.708,20.849C52.708,8.185,44.584,0,32,0S11.292,8.185,11.292,20.849c0,10.556,16.311,31.747,18.175,34.118l2.278,2.928   c-9.542,0.025-17.237,1.38-17.237,3.051c0,1.686,7.836,3.053,17.502,3.053s17.502-1.367,17.502-3.053   c0-1.672-7.704-3.028-17.255-3.051l2.292-2.928C36.396,52.581,52.708,31.39,52.708,20.849z M25.294,20.835   c0-3.604,3.002-6.526,6.706-6.526c3.704,0,6.706,2.922,6.706,6.526S35.704,27.361,32,27.361   C28.296,27.361,25.294,24.439,25.294,20.835z\\"/></g></svg>
          <span class=\\"txt\\">Your location now</span>
        </div>
        <div class=\\"place-name\\">city, country</div>
        <img class=\\"status-icon\\" src=\\"http://openweathermap.org/img/wn/04d.png\\" alt=\\"icon\\"/>
        <div class=\\"status\\">Moonlight</div>
        <div class=\\"temp\\">tempÂ°</div>
        <div class=\\"info-row\\">
          <div class=\\"humidity-info\\">
            <img class=\\"humidity-icon\\" src=\\"test-file-stub\\" alt=\\"water drop\\"/>
            <span class=\\"humidity\\">humidity%</span>
          </div>
          <div class=\\"wind-info\\">
            <img class=\\"wind-icon\\" src=\\"test-file-stub\\" alt=\\"wind\\"/>
            <span class=\\"wind-speed\\">windkm/h</span>
          </div>
          <div class=\\"pressure-info\\">
            <img class=\\"pressure-icon\\" src=\\"test-file-stub\\" alt=\\"pressure\\"/>
            <span class=\\"pressure\\">0.43mBar</span>
          </div>
        </div>
        <div class=\\"btn-rows\\"><div class=\\"btn-row\\">
            <span class=\\"name\\">Temperature</span>
            <span class=\\"value\\">celcius</span>
            <div class=\\"arrow\\">&gt;</div>
          </div><div class=\\"btn-row\\">
            <span class=\\"name\\">Wind speed</span>
            <span class=\\"value\\">km/s</span>
            <div class=\\"arrow\\">&gt;</div>
          </div><div class=\\"btn-row\\">
            <span class=\\"name\\">Source</span>
            <span class=\\"value\\">openWeatherAPI</span>
            <div class=\\"arrow\\">&gt;</div>
          </div></div>
  "
`;
